### PR TITLE
change StateManager and StateChange

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: include/veriblock/state_manager.hpp [hunk 1]
```diff
@@ -114,14 +57,14 @@ class StateManager {
   RepositoryManager database;
 
  public:
-  StateManager(const std::string& name) : database(name) { database.open(); }
+  StateManager(const std::string& name, std::string path = "")
+      : database(path + "\\" + name) {
```
Comment (blackyblack) [inline include/veriblock/state_manager.hpp]: Why this change? We could add separators outside, by the StateManager users.

File: src/popmanager.cpp [hunk 2]
```diff
@@ -84,49 +82,41 @@ bool PopManager::addVTB(const VTB& vtb,
...
-        stateChange.removeBtcBlock(b.getHash());
-      });
+  std::for_each(btcctx.rbegin(), btcctx.rend(), [this](const BtcBlock& b) {
+    btc_->invalidateBlockByHash(b.getHash());
+  });
 }
 
 bool PopManager::addAltProof(const AltProof& payloads,
-                             StateChange& stateChange,
+                             StateChange&,
```
Comment (blackyblack) [inline src/popmanager.cpp]: How about deleting this?

Comment (Mr-Leshiy) [reply src/popmanager.cpp]: It is already deleted in the feature/alt-tree branch

File: include/veriblock/state_manager.hpp (hunk 1 above)
Comment (Mr-Leshiy) [reply include/veriblock/state_manager.hpp]: fixed
